src/viewer.js

```diff
@@ -742,6 +742,10 @@ $.extend( $.Viewer.prototype, $.EventSource.prototype, $.ControlDock.prototype,
             return;
         }
 
+        if (this.source.freeupCanvasMemory) {
```

Comment:
I think this would be better done in TiledImage.destroy; that way it works with multiple images if necessary (as well as in the single-image case).

Reply:
This warning and error seem to be unlikely to happen with multiple images as it uses less canvas memory.
I added TODO comment to implement this for other children of TileSource for now.

Reply:
The multiple image case doesn't necessarily use less memory; you could have multiple ImageTileSources in the same viewer, for instance.

Anyway, moving this code to `TiledImage.destroy` is quite straightforward, and makes the functionality more universal/correct. It will still only affect ImageTileSources, so it's not an expansion into other tile sources.